Frends.MicrosoftSQL.ExecuteProcedure/Frends.MicrosoftSQL.ExecuteProcedure.Tests/NonQueryUnitTests.cs

```diff
@@ -0,0 +1,1029 @@
+        command.CommandText = $@"CREATE PROCEDURE UpdateValue AS update {_tableName} set LastName = 'Edit' where Id = 2";
+        command.ExecuteNonQuery();
+
+        //Delete
+        command.CommandText = $"CREATE PROCEDURE DeleteValue AS delete from {_tableName} where Id = 2";
+        command.ExecuteNonQuery();
+
+        connection.Close();
+        connection.Dispose();
+    }
+
+    [TestCleanup]
+    public void CleanUp()
+    {
+        using var connection = new SqlConnection(_connString);
+        connection.Open();
+        var command = connection.CreateCommand();
+        command.CommandText = $@"IF EXISTS (SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME='{_tableName}') BEGIN DROP TABLE IF EXISTS {_tableName}; END";
+        command.ExecuteNonQuery();
+        command.CommandText = $@"DECLARE @procedureName varchar(500)
+DECLARE cur CURSOR
+      FOR SELECT [name] FROM sys.objects WHERE type = 'p'
+      OPEN cur
+
+      FETCH NEXT FROM cur INTO @procedureName
+      WHILE @@fetch_status = 0
+      BEGIN
+            EXEC('DROP PROCEDURE ' + @procedureName)
+            FETCH NEXT FROM cur INTO @procedureName
+      END
+      CLOSE cur
+      DEALLOCATE cur";
+        command.ExecuteNonQuery();
+        connection.Close();
+        connection.Dispose();
+    }
+
+    [TestMethod]
+    public async Task TestExecuteProcedure_Invalid_Creds_ThrowError()
+    {
+        var input = new Input()
+        {
+            ConnectionString = "Server=127.0.0.1,1433;Database=Master;User Id=SA;Password=WrongPassWord",
+        };
+
+        var options = new Options()
+        {
+            SqlTransactionIsolationLevel = SqlTransactionIsolationLevel.ReadCommitted,
+            CommandTimeoutSeconds = 2,
+            ThrowErrorOnFailure = true
+        };
+
+        var ex = await Assert.ThrowsExceptionAsync<Exception>(() => MicrosoftSQL.ExecuteProcedure(input, options, default));
+        Assert.IsTrue(ex.Message.Contains("SqlException (0x80131904): Login failed for user 'SA'."));
+    }
+
+    [TestMethod]
+    public async Task TestExecuteProcedure_Invalid_Creds_ReturnErrorMessage()
+    {
+        var input = new Input()
+        {
+            ConnectionString = "Server=127.0.0.1,1433;Database=Master;User Id=SA;Password=WrongPassWord",
+        };
+
+        var options = new Options()
+        {
+            SqlTransactionIsolationLevel = SqlTransactionIsolationLevel.ReadCommitted,
+            CommandTimeoutSeconds = 2,
+            ThrowErrorOnFailure = false
+        };
+
+        var result = await MicrosoftSQL.ExecuteProcedure(input, options, default);
+        Assert.IsFalse(result.Success);
+        Assert.IsTrue(result.ErrorMessage.Contains("Login failed for user 'SA'."));
+        Assert.AreEqual(0, result.RecordsAffected);
+    }
+
+    // SqlTransactionIsolationLevel.Unspecified
+    [TestMethod]
+    public async Task TestExecuteProcedure_NonQuery_Unspecified()
+    {
+        var inputInsert = new Input()
```

Comment:
So it seems that you have all transation levels copy-pasted and the only moving part is the isolation level? This project uses MS unit testing, so no parametrized tests like in NUnit, but honestly those should __not__ be just copy-paste. Make them run in a loop like this:
```
var transactionLevels = { none, readcommitted, etc, ... };
foreach(var level in levels) { 
   do the same thing for every enum value
   if test fails include the level in output like
   Assert.Fail("The test failed on level {level}");
};
```

I am not sure if testing every transaction level brings much value for us, but at least doing this in a loop would reduce the amount of code dramatically.

Also, can you check if scalar unit tests contain something like this too (copy-paste that can be easily put in a loop).